tests/self_test.py

```diff
@@ -163,6 +163,44 @@ async def test_self_gatt():
     assert(result == c1.value)
 
 
+# -----------------------------------------------------------------------------
+@pytest.mark.asyncio
+async def test_self_gatt_long_read():
+    # Create two devices, each with a controller, attached to the same link
+    two_devices = TwoDevices()
+
+    # Add some GATT characteristics to device 1
+    characteristics = [
+        Characteristic(
+            f'3A143AD7-D4A7-436B-97D6-5B62C315{i:04X}',
+            Characteristic.READ,
+            Characteristic.READABLE,
+            bytes([x & 255 for x in range(i)])
+        )
+        for i in range(0, 513)
```

Comment:
Why 513?

Reply:
That's because the spec allows GATT characteristic values up to 512 bytes